File: OP2/lab_5/lab_5_c++/lab_5/TNumber.cpp
#include "TNumber.h"

TNumber::TNumber()
{
	number = 0;
}

TNumber::TNumber(int num)
{
	number = num;
}

std::string TNumber::get_str_number()
{
	return std::to_string(number);
}

int TNumber::find_sum()
{
	std::string str = std::to_string(number);
	int res = 0;
	for (int i = 0; i < str.length(); i++) {
		res += str[i] - '0';
	}
	return res;
}

int TNumber::find_first_digit()
{
	return std::to_string(number)[0] - '0';
}

int TNumber::find_last_digit()
{
	std::string str = std::to_string(number);
	return str[str.length() - 1] - '0';
}

File: OP2/lab_5/lab_5_python/TNumber.py
class TNumber:
    def __init__(self, num):
        self._number = num

    def __repr__(self):
        return str(self._number)

    def get_number(self):
        return self._number

    def find_sum(self):
        n = self._number
        sum = 0
        while n > 0:
            digit = n % 10
            sum = sum + digit
            n //= 10
        return sum

    def find_first_digit(self):
        n = str(self._number)
        return int(n[0])

    def find_last_digit(self):
        n = str(self._number)
        return int(n[-1])